mypy/test/teststubtest.py

```diff
@@ -828,6 +828,39 @@ def test_bad_literal(self) -> Iterator[Case]:
             error='WRONG_BOOL_2',
         )
 
+    @collect_cases
+    def test_special_subtype(self) -> Iterator[Case]:
+        yield Case(
+            stub="""
+            b1: bool
+            b2: bool
+            b3: bool
+            """,
+            runtime="""
+            b1 = 0
+            b2 = 1
+            b3 = 2
+            """,
+            error="b3",
+        )
+        yield Case(
+            stub="""
+            from typing_extensions import TypedDict
+
+            class _Options(TypedDict):
+                a: str
+                b: int
+
+            opt1: _Options
+            opt2: _Options
+            """,
+            runtime="""
+            opt1 = {"some": "stuff"}
```

Comment:
This should fail if stubtest was smarter, let's make this a test case where it really should pass
```suggestion
            opt1 = {"a": "stuff", "b": 4}
```

Reply:
Sure, did this to document the behaviour, but just added a third correct case